colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -63,19 +63,28 @@ trait HttpHeaderUtils {
 
   def headers : Seq[(String, String)]
 
-  def getHeader(key : String) : Option[String] = {
-    headers.collectFirst{case (`key`, value) => value}
-  }
 
-  def getHeader(key : String, orElse : String) : String = {
-    headers.collectFirst{case (`key`, value) => value}.getOrElse(orElse)
+  def singleHeader(name: String): Option[String] = {
+    val l = name.toLowerCase
+    headers.collectFirst{ case (`l`, v) => v}
   }
 
-  def getContentLength : Option[Int] = {
-    getHeader(HttpHeaders.ContentLength).map(_.toInt)
+  def multiHeader(name: String): Seq[String] = {
+    val l = name.toLowerCase
+    headers.collect{ case (`l`, v) => v}
   }
 
-  def getEncoding : TransferEncoding = getHeader(HttpHeaders.TransferEncoding).flatMap(TransferEncoding.unapply).getOrElse(TransferEncoding.Identity)
+  //TODO: These vals are probably inefficient, should be generated as the
+  //headers are being parsed.  Benchmark before changing
+
+  /** Returns the value of the content-length header, if it exists.
+   * 
+   * Be aware that lacking this header may or may not be a valid request,
+   * depending if the "transfer-encoding" header is set to "chunked"
+   */
+  val contentLength: Option[Int] = headers.collectFirst{case (HttpHeaders.ContentLength, l) => l.toInt}
+
+  val transferEncoding : TransferEncoding = singleHeader(HttpHeaders.TransferEncoding).flatMap{TransferEncoding.unapply(_)}.getOrElse(TransferEncoding.Identity)
```

Comment:
I'm pretty sure you can turn TransferEncoding.unapply(_) into simply TransferEncoding.unapply.